GeomIO/pFIOServerBounds.F90

```diff
@@ -114,5 +115,18 @@ subroutine initialize(this, grid, field_shape, time_index, rc)
 
    end subroutine initialize
       
+   subroutine finalize(this, rc)
+      class(pFIOServerBounds), intent(inout) :: this
+      integer, intent(out), optional :: rc
+
+      deallocate(this%file_shape)
+      deallocate(this%global_start)
+      deallocate(this%global_count)
+      deallocate(this%local_start)
+
+      _RETURN(_SUCCESS)
+
+   end subroutine finalize
+
```

Comment:
Do we need an explicit finalizer?  Fortran should deallocate these when the object goes out of scope.

Reply:
The way to avoid this would be to convert the `initialize` subroutine into a constructor.

Reply:
The `initialize` is indeed being used in a loop over fields.